test/integration/puppetlabs/services/jruby_pool_manager/jruby_pool_int_test.clj

```diff
@@ -376,7 +300,7 @@
 
          (jruby-core/flush-pool! pool-context)
          ; wait until the flush is complete
-         (await (jruby-agents/get-pool-agent pool-context))
+         (is (jruby-testutils/wait-for-instances (jruby-internal/get-pool pool-context) 1))
```

Comment:
Oh yeah, that's a good point. I was thinking that if you awaited the modify-instance-agent it would have a chance to sneak in before the flush's `send` call, but that can't happen since the flush's `send` would have to have happened before the flush call returns. Yeah, forget what I said about changing the flush call then. Agreed that it shouldn't wait for the whole pool to fill up before returning